File: Cs/Solution/FinalPositionOfSnake.cs
using System.Collections.Generic;

namespace Solution
{
	public partial class Solution
	{
		public int FinalPositionOfSnake(int n, IList<string> commands)
		{
			int pos = 0;
			foreach (var command in commands)
			{
				if (command == "LEFT")
				{
					pos--;
				}
				else if (command == "RIGHT")
				{
					pos++;
				}
				else if (command == "UP")
				{
					pos -= n;
				}
				else
				{
					pos += n;
				}
			}
			return pos;
		}
	}
}

File: Rust/src/solution/final_position_of_snake.rs
use crate::solution::Solution;

impl Solution {
    pub fn final_position_of_snake(n: i32, commands: Vec<String>) -> i32 {
        let mut position = 0;
        for command in commands {
            match command.as_str() {
                "LEFT" => position -= 1,
                "RIGHT" => position += 1,
                "UP" => position -= n,
                "DOWN" => position += n,
                _ => {}
            }
        }
        position
    }
}
